src/mpid/ch4/src/ch4_progress.h

```diff
@@ -101,44 +103,69 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_Progress_vci_shm(int vci)
 }
 
 #undef FUNCNAME
-#define FUNCNAME MPIDI_Progress_test
+#define FUNCNAME MPIDI_Progress_vci
 #undef FCNAME
 #define FCNAME MPL_QUOTE(FUNCNAME)
-MPL_STATIC_INLINE_PREFIX int MPIDI_Progress_test(int flags)
+MPL_STATIC_INLINE_PREFIX int MPIDI_Progress_vci(int hook_flags, int vci_is_nm, int vci)
 {
-    int mpi_errno, vci;
+    int mpi_errno;
     mpi_errno = MPI_SUCCESS;
 
-    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_PROGRESS_TEST);
-    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_PROGRESS_TEST);
-
-#ifdef HAVE_SIGNAL
-    if (MPIDI_CH4_Global.sigusr1_count > MPIDI_CH4_Global.my_sigusr1_count) {
-        MPIDI_CH4_Global.my_sigusr1_count = MPIDI_CH4_Global.sigusr1_count;
-        mpi_errno = MPIDI_check_for_failed_procs();
-        if (mpi_errno)
-            MPIR_ERR_POP(mpi_errno);
+#ifndef MPIDI_CH4_DIRECT_NETMOD
+    if (vci_is_nm) {
+#endif
+        if (hook_flags & MPIDI_PROGRESS_NM) {
+            mpi_errno = MPIDI_Progress_vci_nm(vci);
+            if (mpi_errno != MPI_SUCCESS) {
+                MPIR_ERR_POP(mpi_errno);
+            }
+        }
+#ifndef MPIDI_CH4_DIRECT_NETMOD
```

Comment:
I'd recommend adding this `ifndef` inside the `else` block.  It'd make the code cleaner looking.  It's static inlined, so the compiler will get rid of the branch when there's no shmmods enabled.